src/Angor/Client/Pages/Signatures.razor

```diff
@@ -376,6 +380,60 @@
         }
     }
 
+    private async Task ApproveAllSignatures()
+    {
+        var pendingSignatures = signaturesRequests.Where(s => s.TransactionHex != null && s.AmountToInvest != null && s.TimeApproved == null).ToList();
+        int totalSignatures = pendingSignatures.Count;
+
+        for (int i = 0; i < totalSignatures; i++)
+        {
+            var signature = pendingSignatures[i];
+
+            var operationResult = await notificationComponent.LongOperation(async () =>
+            {
+                var words = await passwordComponent.GetWalletAsync();
```

Comment:
Lok at other pages to see how we have a double method for password request and then we call in to the method that has the spinner

Reply:
I forgot to mention that I did not alter the LongOperation because it comes from your ApproveSignature method. Is there a reason for it there?
or should we try to get rid of it also?